cardano-api/test/Test/Cardano/Api/TxBody.hs

```diff
@@ -0,0 +1,395 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module Test.Cardano.Api.TxBody (tests) where
+
+import           Cardano.Prelude
+
+import           Data.Type.Equality (testEquality)
+import           Hedgehog (PropertyT, evalEither, forAll, property, tripping, (===))
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.Hedgehog (testProperty)
+import           Test.Tasty.TH (testGroupGenerator)
+
+import           Cardano.Ledger.Alonzo.Data (AuxiliaryData (AuxiliaryData))
+import           Cardano.Ledger.Alonzo.TxBody (adHash)
+import           Data.Maybe.Strict (StrictMaybe (SNothing))
+import           Ouroboros.Consensus.Shelley.Eras as Ledger (StandardAlonzo)
+
+import           Cardano.Api
+import           Cardano.Api.Shelley (ProtocolParameters, TxBody (ShelleyTxBody))
+
+import           Gen.Cardano.Api.Typed (genTxBodyContent)
+
+
+-- * Properties
+
+-- | Check that conversion from 'TxBodyContent' to 'TxBody' and back gives
+-- result equivalent to original.
+--
+-- The original randomly-generated data requires
+-- BuildTx/ViewTx type conversion (see View section) and
+-- normalization (see Normalization section).
+--
+-- Roundtrip data requires normalization, too.
+test_roundtrip_TxBody_make_get :: [TestTree]
+test_roundtrip_TxBody_make_get =
+  [ testProperty (show era) $
+    property $ do
+      content <- forAll $ genTxBodyContent era
+      tripping
+        (normalizeContentOriginal $ viewBodyContent content)
+        (\_ -> makeTransactionBody content)
+        (<&> \(TxBody content') -> normalizeContentRoundtrip content')
+  | AnyCardanoEra era <- [minBound..]
+  ]
+
+
+-- | Check that conversion from 'TxBody' to 'TxBodyContent' and back gives
+-- result equivalent to original.
+--
+-- We have no means to generate a random 'TxBody',
+-- so we start from 'TxBodyContent' and assume 'TxBody' made from it
+-- is arbitrary enough.
+--
+-- The original randomly-generated data requires
+-- ViewTx/BuildTx type conversion (see Build section).
+--
+-- No normalization is needed here, because 'TxBody' keeps less information
+-- on type and value level than 'TxBodyContent'.
+-- For instance, no special /None/ values.
+test_roundtrip_TxBody_make_get_make :: [TestTree]
+test_roundtrip_TxBody_make_get_make =
+  [ testProperty (show era) $
+    property $ do
+
+      -- generate a TxBodyContent to use as a seed
+      content0 <- forAll $ genTxBodyContent era
+      let TxBodyContent{txProtocolParams} = content0 -- keep for re-building
+
+      -- make an "original" TxBody
+      body1 <- evalEither $ makeTransactionBody content0
+
+      -- convert TxBody back to TxBodyContent
+      let TxBody content1 = body1
+
+      -- prepare content1 for making another TxBody
+      let content2 = buildBodyContent txProtocolParams content1
+
+      -- make another TxBody -- roundtrip
+      body2 <- evalEither $ makeTransactionBody content2
+
+      assertEqBodies body1 body2
+
+  | AnyCardanoEra era <- [minBound..]
+  ]
+
+
```

Comment:
A comment like: "The auxiliary data hashes are different (and therefore must be ignored) due to the representational differences between the ledger's tx body and the api's txbody" would be useful

Reply:
Extended docs a bit.